### PR TITLE
CDAP-13380 CDAP Cloud Logging Integration

### PR DESCRIPTION
**Pull request by anwar6953**
The first commit is a revert of a revert: https://github.com/caskdata/cdap/pull/10305, where the original PR was: https://github.com/caskdata/cdap/pull/10258.

I reverted it because it had a bug, which presented itself as high CPU usage, and everything being much slower. For instance, programs took about twice as long to launch and run.
The second commit fixes the bug.

The third commit implements the functionality for distributed CDAP.

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/remote/DistributedLogProcessor.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Iterator;
+
+/**
+ * The {@link RemoteExecutionLogProcessor} used in Distributed mode.
+ */
+public class DistributedLogProcessor implements RemoteExecutionLogProcessor {
+
+  @Override
+  public void process(Iterator<byte[]> loggingEventBytes) {
+    // TODO (CDAP-13380): implement distributed log processor
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/remote/DistributedLogProcessor.java]: Still a TODO??

Comment (anwar6953) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/remote/DistributedLogProcessor.java]: Ah, the implementation doesn't use this class. I will remove this class, as it is unused.

**Comment by anwar6953** [comment]
Build: https://builds.cask.co/browse/CDAP-RUT1538-5

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/AbstractLogPublisher.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        if (failures == 0) {
+          failureStartTime = System.currentTimeMillis();
+        }
+
+        long sleepMillis = retryStrategy.nextRetry(++failures, failureStartTime);
+        if (sleepMillis < 0) {
+          buffer.clear();
+          failures = 0;
+
+          // Log using the status manager
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/AbstractLogPublisher.java]: This comment is not right.

**Review by chtyim — COMMENTED** [review]
Just one comment about a misplaced comment. rest LGTM.

**Comment by anwar6953** [comment]
Addressed the comment, RUT build is green.
However, ITN is not passing. I'm currently investigating.

**Comment by anwar6953** [comment]
https://builds.cask.co/browse/CDAP-RUT1538-7

**Comment by anwar6953** [comment]
Removed the latest refactoring commit as it was causing ITN failures. ITN passes now.
The build is also failing on flaky tests: https://builds.cask.co/browse/CDAP-RUT1538-10

**Pull request merged**